### DISCUSSION TITLE
How does command autocompletion work?

### DISCUSSION THREAD

**Discussion by DragonZV** [marlonrichert/zsh-autocomplete#386] [Q&A] [answered]
I'm not sure what kind of isssue to make, so I'm doing feature request.

How does command autocompletion work? I have a slightly complex version of pacman I made, and I'm wondering how I can have zsh-autocomplete autocomplete my flags or packages.

![image](https://user-images.githubusercontent.com/70718540/146152945-b790508e-f7ce-4c10-9c6c-06b08550071d.png)

How does it know what flags are applicable for this command?

![image](https://user-images.githubusercontent.com/70718540/146153269-96a6f9fd-a8b1-457b-96dd-bf81962a538f.png)

How do I have my command have my command's parameters autocomplete with Arch Linux packages?

**Comment by marlonrichert** [comment, marked as answer]
> I'm not sure what kind of isssue to make, so I'm doing feature request.

That's what discussions are for. 🙂  I've converted your issue to a discussion.

> How does command autocompletion work? I have a slightly complex version of pacman I made, and I'm wondering how I can have zsh-autocomplete autocomplete my flags or packages.
> How does it know what flags are applicable for this command?
> How do I have my command have my command's parameters autocomplete with Arch Linux packages?

[`pacman`'s own ArchLinux package](https://archlinux.org/packages/core/x86_64/pacman/) includes [a Zsh completion function for itself](https://gitlab.archlinux.org/pacman/pacman/-/blob/master/scripts/completion/zsh_completion.in). This file is installed as `/usr/share/zsh/site-functions/_pacman` when the `pacman` package is installed.

**Comment by DragonZV** [comment]
I see, thanks!
